app/src/main/java/org/cru/godtools/ui/dashboard/tools/ToolsFragment.kt

```diff
@@ -2,24 +2,101 @@ package org.cru.godtools.ui.dashboard.tools
 
 import android.os.Bundle
 import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.ConcatAdapter
+import androidx.recyclerview.widget.RecyclerView
 import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import org.ccci.gto.android.common.androidx.recyclerview.widget.addLayout
 import org.ccci.gto.android.common.sync.swiperefreshlayout.widget.SwipeRefreshSyncHelper
 import org.cru.godtools.R
 import org.cru.godtools.databinding.DashboardToolsFragmentBinding
+import org.cru.godtools.download.manager.GodToolsDownloadManager
 import org.cru.godtools.fragment.BasePlatformFragment
+import org.cru.godtools.model.Tool
+import org.cru.godtools.model.Translation
+import org.cru.godtools.ui.tooldetails.startToolDetailsActivity
+import org.cru.godtools.ui.tools.ToolsAdapter
+import org.cru.godtools.ui.tools.ToolsAdapterCallbacks
+import org.cru.godtools.ui.tools.ToolsAdapterViewModel
 
 @AndroidEntryPoint
-class ToolsFragment : BasePlatformFragment<DashboardToolsFragmentBinding>(R.layout.dashboard_tools_fragment) {
+class ToolsFragment :
+    BasePlatformFragment<DashboardToolsFragmentBinding>(R.layout.dashboard_tools_fragment),
+    ToolCategoriesAdapter.Callbacks,
+    ToolsAdapterCallbacks {
+    @Inject
+    internal lateinit var downloadManager: GodToolsDownloadManager
 
-    private val dataModel: ToolsFragmentDataModel by viewModels()
+    // region Data Model
+    private val dataModel: ToolsCategoryDataModel by viewModels()
```

Comment:
the data model class should be named `ToolsFragmentDataModel`, not `ToolsCategoryDataModel`